byway/src/main/java/com/google/sps/data/Trip.java

```diff
@@ -74,6 +74,15 @@ public Trip(
     return Collections.unmodifiableList(this.route);
   }
 
+  /**
+   * Sets the route of the trip with a list of stops and destinations as plain text.
+   *
+   * @param route list of user-selected stops and destinations
+   */
+  public void setRoute(Collection<String> route) {
+    this.route = new ArrayList<>(route);
```

Comment:
keep `route` as final and  use `this.route.clear(); this.route.addAll(route);` Using clear + assign here is nice because it potentially avoids allocating a new array, and lets you keep the field as final.

Reply:
Changed. Thanks for explaining `final` in mutable classes.